WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -80,6 +90,54 @@ - (XCElementSnapshot *)fb_lastSnapshot
   return self.lastSnapshot;
 }
 
+- (XCElementSnapshot *)fb_snapshotWithAttributes {
+  if (![XCElementSnapshot.class respondsToSelector:@selector(snapshotAttributesForElementSnapshotKeyPaths:)]) {
+    return nil;
+  }
+  [self resolve];
+  
+  id defaultParameters = [[XCAXClient_iOS sharedClient] defaultParameters];
+  
+  // Names of the properties to load. There won't be lazy loading for missing properties,
+  // thus missing properties will lead to wrong results
+  NSArray<NSString *> *propertyNames = @[
+                                         @"identifier",
+                                         @"value",
+                                         @"label",
+                                         @"frame",
+                                         @"enabled",
+                                         @"elementType"
+                                         ];
+  
+  NSSet *attributes = [XCElementSnapshot snapshotAttributesForElementSnapshotKeyPaths:propertyNames];
+  
+  NSArray *axAttributes = XCAXAccessibilityAttributesForStringAttributes(attributes);
+  if (![axAttributes containsObject:FB_XCAXAIsVisibleAttribute]) {
+    axAttributes = [axAttributes arrayByAddingObject:FB_XCAXAIsVisibleAttribute];
+  }
+  
+  __block XCElementSnapshot *snapshotWithAttributes = nil;
+  dispatch_group_t resolveGroup = dispatch_group_create();
+  dispatch_group_enter(resolveGroup);
+  
+  id<XCTestManager_ManagerInterface> proxy = [FBXCTestDaemonsProxy testRunnerProxy];
+  
+  [proxy _XCT_snapshotForElement:self.lastSnapshot.accessibilityElement
+                      attributes:axAttributes
+                      parameters:defaultParameters
+                           reply:^(XCElementSnapshot *snapshot, NSError *error) {
+                             if (error != nil) {
+                               dispatch_group_leave(resolveGroup);
+                               return;
+                             }
+                             snapshotWithAttributes = snapshot;
+                             dispatch_group_leave(resolveGroup);
+                           }];
+  
+  dispatch_group_wait(resolveGroup, dispatch_time(DISPATCH_TIME_NOW, 5 * NSEC_PER_SEC));
```

Comment:
should we verify if this times out?

Reply:
I can add a check to make it more verbose, but in the end if it times out it would return `nil` anyways and use the other way of getting the snapshot without the attributes.

Reply:
we can add a simple debug log there. Just in case...